src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const toBeRemoved of action.keysToRemove) {
+          if (toBeRemoved in state) {
+            delete state[toBeRemoved];
+          }
+        }
+        break;
 
+      case 'clear':
+        for (const stateKey in state) {
```

Comment:
just `key` is enough

Reply:
It' really enough  :))

All your remarks are much appreciated ! Make me focuced on all small but critical points in development !